src/bunit.core/Rendering/TestRenderer.cs

```diff
@@ -117,6 +118,27 @@ public IReadOnlyList<IRenderedComponentBase<TComponent>> FindComponents<TCompone
 		where TComponent : IComponent
 		=> FindComponents<TComponent>(parentComponent, int.MaxValue);
 
+
+	/// <inheritdoc />
+	public void DisposeComponents()
+	{
+		var disposeTask = Dispatcher.InvokeAsync(() =>
+		{
+			foreach (var root in rootComponents)
+			{
+				root.Detach();
+			}
+		});
+
+		if (!disposeTask.IsCompleted)
+		{
+			disposeTask.GetAwaiter().GetResult();
+		}
```

Comment:
```suggestion
		// The dispatcher will always return a completed task, 
		// since calling root.Detach does not trigger any (async) Dispose
		// operations directly on the components. It happens
		// indirectly after a re-render, where the components
		// are removed from the render tree.
		Dispatcher.InvokeAsync(() =>
		{
			foreach (var root in rootComponents)
			{
				root.Detach();
			}
		});		
```

Lets simplify this and leave an explanation for our future selves :)

Reply:
Well the explanation is not completely true and I would like not to delete the `diposeTas.GetAwaiter().GetResult()` stuff as we need this for throwing an expcetion in `TestContextBase.DisposeComponents` if our component under test will itself throw the exception in `Dispose`.

The sync version will lead to a direct update of the renderer...only `DisposeAsync` is special

Reply:
Though I will add the comment for the async version for our future self :D